core/src/main/java/org/elasticsearch/http/netty/NettyHttpRequest.java

```diff
@@ -69,6 +69,14 @@ public NettyHttpRequest(org.jboss.netty.handler.codec.http.HttpRequest request,
     @Override
     public Method method() {
         HttpMethod httpMethod = request.getMethod();
+
+        if (request.headers() != null) {
+            String methodOverride = request.headers().get("X-HTTP-Method-Override");
+            if (methodOverride != null) {
+                httpMethod = HttpMethod.valueOf(methodOverride);
+            }
+        }
+
```

Comment:
Agreed. Actually this PATCH is at least not safe enough.
Someone could send a `GET` request to `/_all` with `X-HTTP-Method-Override: DELETE` and remove everything!

At least, as stated in [RFC 2616 - HTTP 1.1 section 9.1.1](http://www.ietf.org/rfc/rfc2616.txt):

> Implementors should be aware that the software represents the user in
   their interactions over the Internet, and should be careful to allow
   the user to be aware of any actions they might take which may have an
   unexpected significance to themselves or others.
>
>   In particular, the convention has been established that the GET and
   HEAD methods SHOULD NOT have the significance of taking an action
   other than retrieval. These methods ought to be considered "safe".
   This allows user agents to represent other methods, such as POST, PUT
   and DELETE, in a special way, so that the user is made aware of the
   fact that a possibly unsafe action is being requested.

At least I should only accept this if `httpMethod == POST`.